backend/layers/processing/process_seurat.py

```diff
@@ -61,12 +63,19 @@ def process(self, dataset_version_id: DatasetVersionId, artifact_bucket: str, da
             self.logger.info("Skipping Seurat conversion")
             return
 
+        # Download h5ad locally
         labeled_h5ad_filename = "local.h5ad"
-
         key_prefix = self.get_key_prefix(dataset_version_id.id)
         object_key = f"{key_prefix}/{labeled_h5ad_filename}"
         self.download_from_s3(artifact_bucket, object_key, labeled_h5ad_filename)
 
+        # Convert the citation from h5ad to RDS
+        adata = anndata.read_h5ad(labeled_h5ad_filename)
+        if "citation" in adata.uns:
+            adata.uns["citation"] = adata.uns["citation"].replace(".h5ad", ".rds")
```

Comment:
let's make use of the [dataset id regex](https://github.com/chanzuckerberg/single-cell-data-portal/blob/2f655e203430d69634f8b9885d6780a747c8e4ae/backend/common/utils/regex.py#L5) so that we don't get caught by a doi that has `.h5ad` in it.